### PR TITLE
Add a tiny JS world OOP version

### PR DESCRIPTION
**Pull request by Vitaminvp**
### a tiny JS world OOP version

[Demo](https://vitaminvp.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/Vitaminvp/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/Vitaminvp/a-tiny-js-world-OOP/index.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    addFriends(...args){
+        this.friends = [...args];
+    }
+	printCreature(){
+		return ['species', 'gender', 'name', 'legs', 'hands', 'saying']
+				  .map(key => `<strong>${this[key]}</strong>`)
+				  .concat(this.friends.map(friend => `<em>${friend['name']}</em>`))
+				  .join('; ');
+	}
+}
```
Comment (OleksiyRudenko) [inline submissions/Vitaminvp/a-tiny-js-world-OOP/index.js]: Please, fix code formatting.
We write code for other people, firsthand. One of the ways to take care of colleagues and peers is to follow conventions on code formatting.
The code below is formatted almost perfectly.

File: submissions/Vitaminvp/a-tiny-js-world-OOP/index.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+class SuperHero extends Creature {
+    constructor(prototipe1, prototipe2, name, legs, hands) {
+      super(null, null, name, legs, hands);
+	  this.species = prototipe1.species + '-' + prototipe2.species;
+	  this.gender  = prototipe1.gender;
```
Comment (OleksiyRudenko) [inline submissions/Vitaminvp/a-tiny-js-world-OOP/index.js]: Please, fix indentation.

Comment (OleksiyRudenko) [reply submissions/Vitaminvp/a-tiny-js-world-OOP/index.js]: It is also not really OOP-compliant borrowing methods/properties from instances.
If do not find different way just skip fixing this part as it is optional.

File: submissions/Vitaminvp/a-tiny-js-world-OOP/index.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+// ======== OUTPUT ========
+/* Use print(message) for output.
+   Default tag for message is <pre>. Use print(message,'div') to change containing element tag.
+
+   Message can contain HTML markup. You may also tweak index.html and/or styles.css.
+   However, please, REFRAIN from improving visuals at least until your code is reviewed
+   so code reviewers might focus on a single file that is index.js.
+   */
+   [woman, man, cat, dog, catWoman, dogMan].map(creature => print(creature.printCreature()));
```
Comment (OleksiyRudenko) [inline submissions/Vitaminvp/a-tiny-js-world-OOP/index.js]: Fix indentation.

**Comment by OleksiyRudenko** [comment]
@Vitaminvp Please, check yourself against the list below.

Must have:
 * [ ] Implement a base class to inherit from
 * [ ] Employ default parameters
 * [ ] Each species is represented with its own class
 * [ ] No need to specify species at instantiation
 * [ ] Classes for species that do not have hands by natural design
       do not consequently have `hands` or any equivalent property
 * [ ] All inhabitants are stored in a container (array or object)
 * [ ] JS native features are intensively employed (`const`, `let`, `Array.map|join|forEach|...`, etc)
 * [ ] [OOP, SOLID and DRY](https://github.com/OleksiyRudenko/a-tiny-JS-world/blob/master/README.md#learn-on-your-own)
       principles are intensively employed
 
Optional level up (not required to implement):
 * [ ] Friends list is a list of objects refs rather than names (strings)
 * [ ] Cat-woman class is built employing composition rather than inheritance only
 * [ ] `toString`
 
[ES6 classes cheat-sheet](https://gist.github.com/OleksiyRudenko/672d39b08d9d0da4e179aca49876c58b)

Please, confirm all items above are positive. Please, ask a question if anything above requires clarification.

**Comment by Vitaminvp** [comment]
I changed something, hope not too much.

**Comment by Vitaminvp** [comment]
Ok. I think that:
a) common for every inhabitant:  **name, gender.**
b) unique for humans: legs, hands
c) unique for dogs: say

**Comment by OleksiyRudenko** [comment]
@Vitaminvp Unique is something someone has and others do not. Common are things that everyone has.
How did you decide that gender is a unique property and species is not?

**Comment by OleksiyRudenko** [comment]
@Vitaminvp Perfectly correct! Please, fix your code accordingly.

**Comment by Vitaminvp** [comment]
What about this one?

File: submissions/Vitaminvp/a-tiny-js-world-OOP/index.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+class Woman extends HumanBeing {
+    constructor(name, gender='female', species, say = 'Love saves the world.', legs, hands) {
+        super(name, gender, species, say, legs, hands);
+    }
+}
+class CatWoman extends Woman {
+    constructor(name, gender, species, say) {
+        super(name, gender, species, say);
+    }
```
[the remark this answers is not in the document]
Comment (Vitaminvp) [reply submissions/Vitaminvp/a-tiny-js-world-OOP/index.js]: Not sure about correctness, but it works, I hope.

File: submissions/Vitaminvp/a-tiny-js-world-OOP/index.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+class Woman extends HumanBeing {
+    constructor(name, say = 'Love saves the world.', legs, hands) {
+        super(name, 'female', say, legs, hands);
+    }
+}
+class CatWoman extends Woman {
+    constructor(name, say) {
+        super(name, say);
+        this.say = new Cat().say;
+    }
```
Comment (OleksiyRudenko) [inline submissions/Vitaminvp/a-tiny-js-world-OOP/index.js]: Nice try. Here is how it reads: let's create an **instance** of `Cat` and borrow a method for a **class**. Not really good.
The better way would be to borrow a method from `Cat` class prototype chain.
But let's keep this as is for now.

**Pull request merged**